src/Native/Runtime/thread.cpp

```diff
@@ -398,7 +387,9 @@ void Thread::Destroy()
     }
 #endif // !PROJECTN
 
-    RedhawkGCInterface::ReleaseAllocContext(GetAllocContext());
+    gc_alloc_context* m_alloc_context = GetAllocContext();
+    dead_threads_non_alloc_bytes += m_alloc_context->alloc_limit - m_alloc_context->alloc_ptr;
```

Comment:
This needs to be interlocked operation, I think.

Reply:
I took inspiration from https://github.com/dotnet/coreclr/blob/61dfc51ae30a9a2dbdb637fa200af85ea0435dc5/src/vm/threads.cpp#L2987-L2989
which doesn't either and is talking about the threadstore lock which is taken here https://github.com/dotnet/corert/blob/4357f9b8be3f5856ddce285c2534b6f3b08752bd/src/Native/Runtime/threadstore.cpp#L176-L180 just before `Destroy`ing the thread so i've assumed this doesn't need interlocked, correct?

Reply:
Ok, you are right.